search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityService.kt

```diff
@@ -0,0 +1,100 @@
+package com.egm.stellio.search.service
+
+import com.egm.stellio.search.model.AttributeInstanceResult
+import com.egm.stellio.search.model.FullAttributeInstanceResult
+import com.egm.stellio.search.model.TemporalEntityAttribute
+import com.egm.stellio.shared.model.CompactedJsonLdAttribute
+import com.egm.stellio.shared.model.CompactedJsonLdEntity
+import com.egm.stellio.shared.util.*
+import org.springframework.stereotype.Service
+import java.net.URI
+
+typealias SimplifiedTemporalAttribute = Map<String, Any>
+
+@Service
+class TemporalEntityService {
```

Comment:
I've just spent a moment trying to understand it fully and it remains hard to read and understand.

At a minimum, add some doc to explain what is going on in each function (since this is not obvious by the signatures of the functions).

I also think there are too many mutable collections in this class...

Reply:
I'll try to improve it and add some doc.

Reply:
I've pushed a commit with some improvements.